src/components/ManageTransactions/index.js

```diff
@@ -0,0 +1,132 @@
+import React, { useState } from 'react';
+import { useEffect } from 'react';
+import { useTransactionContext } from '../../context/globalState';
+import Button from '../Button';
+import TransactionItem from '../TransactionItem';
+import { StyledTransactionLabel, StyledTransactionInput } from '../TransactionForm/styled';
+import { StyledManageTransactionsWrapper, StyledManageTransactions, StyledManageTransactionsForm } from './styled';
+
+const Categories = () => {
+  const [type, setType] = useState('Income');
+  const [category, setCategory] = useState('All');
+  const [startDate, setStartDate] = useState(new Date().toISOString().split('T')[0]);
+  const [endDate, setEndDate] = useState(new Date().toISOString().split('T')[0]);
+  const [sortType, setSortType] = useState('dateDesc');
+  const [filteredTransactions, setFilteredTransactions] = useState([]);
+  
+  const { transactions, incomeCategories, expenseCategories } = useTransactionContext();
+  const categories = type === 'Income' ? incomeCategories : expenseCategories;
+
+  useEffect(() => {
+    filterTransactions();
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [transactions]);
+
+  const typeHandler = (type) => {
+    setType(type);
+    setCategory('All');
+  };
+
+  const filterTransactions = () => {
+    if (category === 'All') {
+      setFilteredTransactions(transactions.filter((item) => {
+        return type === item.type && item.date >= startDate && item.date <= endDate;
+      }).sort((a, b) => sortTransactions(a, b, sortType)));
+    } else {
+      setFilteredTransactions(transactions.filter((item) => {
+        return category === item.category && item.date >= startDate && item.date <= endDate;
+      }).sort((a, b) => sortTransactions(a, b, sortType)));
+    }
+  };
+
+  const sortTransactions = (a, b, sortType) => {
+    switch(sortType) {
+      case 'dateAsc':
+        return a.date < b.date ? 1 : a.date > b.date ? -1 : 0;
+      case 'dateDesc':
+        return a.date < b.date ? -1 : a.date > b.date ? 1 : 0;
+      case 'amountAsc':
+        return a.amount - b.amount;
+      case 'amountDesc':
+        return b.amount - a.amount;
+      default:
+        return;
+    }
+  };
+
+  const onSubmit = (evt) => {
+    evt.preventDefault();
+    filterTransactions();
+  };
+
+  return (
+    <StyledManageTransactionsWrapper>
+      <StyledManageTransactionsForm onSubmit={(evt) => onSubmit(evt)}>
+        <div>
+          <div>
+            <div>
```

Comment:
wondering about the importance of the devs separating the two inputs here... can we remove them, rendering the two inputs next to each other and style the parent div to reach the same result?